nifi-nar-bundles/nifi-adx-bundle/nifi-adx/src/main/java/org/apache/nifi/adx/AzureAdxSourceConnectionService.java

```diff
@@ -0,0 +1,186 @@
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import com.microsoft.azure.kusto.data.Client;
+import com.microsoft.azure.kusto.data.ClientFactory;
+import com.microsoft.azure.kusto.data.auth.ConnectionStringBuilder;
+
+import java.util.List;
+import static org.apache.nifi.adx.NiFiVersion.NIFI_SOURCE;
+
+@Tags({"Azure", "ADX", "Kusto", "ingest", "azure"})
+@CapabilityDescription("Sends batches of flow file content or stream flow file content to an Azure ADX cluster.")
+@ReadsAttributes({
+        @ReadsAttribute(attribute = "AUTH_STRATEGY", description = "The strategy/method to authenticate against Azure Active Directory, either 'application' or 'managed_identity'."),
+        @ReadsAttribute(attribute = "APP_ID", description = "Specifies Azure application id for accessing the ADX-Cluster."),
+        @ReadsAttribute(attribute = "APP_KEY", description = "Specifies Azure application key for accessing the ADX-Cluster."),
+        @ReadsAttribute(attribute = "APP_TENANT", description = "Azure application tenant for accessing the ADX-Cluster."),
+        @ReadsAttribute(attribute = "CLUSTER_URL", description = "Endpoint of ADX cluster. This is required only when streaming data to ADX cluster is enabled."),
+})
+public class AzureAdxSourceConnectionService extends AbstractControllerService implements AdxSourceConnectionService {
+
+    private static final String KUSTO_STRATEGY_APPLICATION = "application";
+
+    private static final String KUSTO_STRATEGY_MANAGED_IDENTITY = "managed_identity";
+
+    public static final PropertyDescriptor KUSTO_AUTH_STRATEGY = new PropertyDescriptor
+            .Builder().name(AzureAdxConnectionServiceParameter.AUTH_STRATEGY.name())
+            .displayName(AzureAdxConnectionServiceParameter.AUTH_STRATEGY.getParamDisplayName())
+            .description(AzureAdxConnectionServiceParameter.AUTH_STRATEGY.getDescription())
+            .required(false)
+            .defaultValue(KUSTO_STRATEGY_APPLICATION)
+            .allowableValues(KUSTO_STRATEGY_APPLICATION, KUSTO_STRATEGY_MANAGED_IDENTITY)
+            .build();
+
+    public static final PropertyDescriptor APP_ID = new PropertyDescriptor
+            .Builder().name(AzureAdxConnectionServiceParameter.APP_ID.name())
+            .displayName(AzureAdxConnectionServiceParameter.APP_ID.getParamDisplayName())
+            .description(AzureAdxConnectionServiceParameter.APP_ID.getDescription())
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor APP_KEY = new PropertyDescriptor
+            .Builder().name(AzureAdxConnectionServiceParameter.APP_KEY.name())
+            .displayName(AzureAdxConnectionServiceParameter.APP_KEY.getParamDisplayName())
+            .description(AzureAdxConnectionServiceParameter.APP_KEY.getDescription())
+            .required(true)
```

Comment:
This property should be marked as sensitive.
```suggestion
            .required(true)
            .sensitive(true)
```